src/main/java/com/exasol/repository/ReleaseLetterParser.java

```diff
@@ -0,0 +1,107 @@
+package com.exasol.repository;
+
+import static com.exasol.ReleaseRobotConstants.DATE_REGEX;
+import static com.exasol.ReleaseRobotConstants.VERSION_REGEX;
+
+import java.time.LocalDate;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.exasol.repository.ReleaseLetter.Builder;
+
+public final class ReleaseLetterParser {
+    public static final String TICKET_NUMBER_REGEX = "#[1-9]\\d*\\b";
+
+    private ReleaseLetterParser() {
+        // prevent instantiation
+    }
+
+    /**
+     * Create a new {@link ReleaseLetter} from a string.
+     *
+     * @param fileName name of the release letter file
+     * @param content file's content as a string
+     */
+    public static ReleaseLetter parseReleaseLetterContent(final String fileName, final String content) {
```

Comment:
Pass directories and file names as Path (segment) where possible. Only if you really want to do string arithmetic use strings.

Reply:
Turn the Parser into a dynamic class. Construct with path to file that needs to be parsed.

Reply:
It doesn't make sense to use a Path here because an external library I use takes String as a path.  :(

Reply:
Okay.